lib/mixins/property-effects.html

```diff
@@ -462,37 +456,43 @@
    *
    * @param {Object} model Prototype or instance
    * @param {Object} note Annotation note returned from Annotator
-   * @param {number} index Index into `__templateNodes` list of annotated nodes that the
-   *   note applies to
+   * @param {number} index Index into `nodeList` that the binding applies to
    * @param {Object=} dynamicFns Map indicating whether method names should
    *   be included as a dependency to the effect.
    * @private
    */
-  function addBindingEffect(model, note, index, dynamicFns) {
-    for (let i=0; i<note.parts.length; i++) {
-      let part = note.parts[i];
-      if (part.signature) {
-        addMethodBindingEffect(model, note, part, index, dynamicFns);
-      } else if (!part.literal) {
-        if (note.kind === 'attribute' && note.name[0] === '-') {
-          console.warn('Cannot set attribute ' + note.name +
+  function addBinding(constructor, templateInfo, nodeInfo, kind, target, parts, literal) {
+    // Create binding metadata and add to nodeInfo
+    nodeInfo.bindings = nodeInfo.bindings || [];
+    let binding = { kind, target, parts, literal, isCompound: (parts.length !== 1) };
+    let index = templateInfo.nodeInfoList.length;
+    nodeInfo.bindings.push(binding);
+    // Add 2-way binding listener metadata to templateInfo
+    if (shouldAddListener(binding)) {
+      addAnnotatedListener(templateInfo, index, binding);
+    }
+    // Add "propagate" property effects to templateInfo
+    for (let i=0; i<binding.parts.length; i++) {
+      let part = binding.parts[i];
+      if (!part.literal) {
+        if (binding.kind === 'attribute' && binding.target[0] === '-') {
+          console.warn('Cannot set attribute ' + binding.target +
             ' because "-" is not a valid attribute starting character');
         } else {
-          model._addPropertyEffect(Polymer.Path.root(part.value), TYPES.PROPAGATE, {
-            fn: runBindingEffect,
-            info:  {
-              kind: note.kind,
-              index: index,
-              name: note.name,
-              propertyName: note.propertyName,
-              value: part.value,
-              isCompound: note.isCompound,
-              compoundIndex: part.compoundIndex,
-              event: part.event,
-              customEvent: part.customEvent,
-              negate: part.negate
+          let dependencies = part.dependencies;
+          let info = { index, binding, part, evaluator: constructor };
+          part.compoundIndex = i;
+          for (let j=0; j<dependencies.length; j++) {
+            let trigger = dependencies[j];
+            if (typeof trigger == 'string') {
```

Comment:
Bookmark to consider if we want to assume we should `parseArg` the `trigger` if it's a string.